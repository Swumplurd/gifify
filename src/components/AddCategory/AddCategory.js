import {useState} from 'react';
import PropTypes from 'prop-types'


export const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 0) {            
            setCategories(cats => [inputValue, ...cats]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <div className="mb-3">
                <p>{inputValue}</p>
                <label htlmfor="categorias" className="form-label">Buscar Categorias</label>
                <input type="text" value={inputValue} onChange={handleInputChange} className="form-control shadow" id="categorias" aria-describedby="categoriasHelp"/>
                <div id="categoriasHelp" className="form-text">Ingresa el nombre de una nueva categoria...</div>
            </div>
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}