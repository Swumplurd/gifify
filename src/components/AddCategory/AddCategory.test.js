import { shallow } from "enzyme";
import { AddCategory } from "./AddCategory";

describe('Pruebas en el componente AddCategory', () => {
    const setCategories = jest.fn()
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    })

    test('Debe de coincidir con el snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('Debe de cambiar el input', () => {
        const input = wrapper.find('input');
        const value = 'Hola Prro'

        input.simulate('change', {target: { value: value }})
        expect(wrapper.find('p').text()).toBe(value)
    })

    test('Se espera que no se postee informacion en el evento submit', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){}})

        expect(setCategories).not.toHaveBeenCalled();
    })

    test('debe de llamar setCategories y limpiar la caja de texto', () => {
        const value = 'Hola'

        const input = wrapper.find('input');
        input.simulate('change', {target: { value: value }})

        const form = wrapper.find('form');
        form.simulate('submit', {preventDefault(){}})

        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledWith( expect.any(Function));

        expect(wrapper.find('input').prop('value')).toBe('')
    })
    
    
    
    
})
