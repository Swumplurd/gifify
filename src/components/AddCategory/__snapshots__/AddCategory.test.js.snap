// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AddCategory Debe de coincidir con el snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="mb-3"
  >
    <p />
    <label
      className="form-label"
      htlmfor="categorias"
    >
      Buscar Categorias
    </label>
    <input
      aria-describedby="categoriasHelp"
      className="form-control shadow"
      id="categorias"
      onChange={[Function]}
      type="text"
      value=""
    />
    <div
      className="form-text"
      id="categoriasHelp"
    >
      Ingresa el nombre de una nueva categoria...
    </div>
  </div>
</form>
`;
